redis/asyncio/sentinel.py

```diff
@@ -116,12 +175,17 @@ def __init__(self, service_name, sentinel_manager, **kwargs):
         )
         self.is_master = kwargs.pop("is_master", True)
         self.check_connection = kwargs.pop("check_connection", False)
+        self.proxy = SentinelConnectionPoolProxy(
```

Comment:
Well, conceptually proxies are used to add an additional functionality before or after we want to access the original object. But here proxy are used just to override super class methods that breaks the functionality of child. 

Proxy here is the workaround, to hide a problem with inheritance that we have. We have to either fix it in ConnectionPool so it can be inheritable, or doesn't inherit this at all